### ISSUE TITLE
Logging doesn't work propertly if Robot is running on thread

### ISSUE THREAD

**Issue by arnoldh14** [robotframework/robotframework#5138]
When I call my test suite/test case via the normal command line or run it in vscode, the Log function behaves as normal. When I call the same test suite/test case using the robot.run(...) api, it does not log the Log function details properly in the report.

Env:
Python: 3.12
Robot framework: 7.0
Windows 11

As an example:
file: test.robot
```
*** Variables ***
${test_notes}   'default notes text'

*** Test Cases ***
Test-P1
    [Documentation]  Simple log test case
    Log To Console   A message only to the console
    Log   Report and console message ${test_notes}   console=${True}
```

When I run this test in the cmd or with vscode, I get the correct output (default message from the variable) i.e.
![image](https://github.com/robotframework/robotframework/assets/65655682/2eba533b-b967-4783-9ab9-b90ec861a872)

When I run this in my application with the api and override the default message with the variable argument i.e.
`ret_code = run('./tests/test.robot', test=f"Test-P1", variable=f"test_notes:Notes from python...", loglevel='INFO', outputdir='./results')`

I get the following result:
![image](https://github.com/robotframework/robotframework/assets/65655682/5816fcff-e0ff-4915-988d-2b600eac0866)

notice that the INFO message is not displayed anymore. I can only get it to display when I set the loglevel=TRACE. It does server as a workaround for now, but is not desired.
![image](https://github.com/robotframework/robotframework/assets/65655682/ff372660-94bb-457c-85d5-ae6365272ca1)

This seems like a bug.
Thanks in advance.

**Comment by arnoldh14** [comment]
Yes - it does. I spin up a worker thread else my UI will "hang up".

**Comment by arnoldh14** [comment]
Ok. I confirmed what you were saying; The logging works as expected when running the test case in the main thread. The UI however locks up - as predicted. 
I see two work-arounds for now: 
1. Set the loglevel=TRACE in the call Robot.run(..., loglevel=TRACE, ...), and use Log as is - hopefully this does not mess up the output.xml file?
2. Calling ${log_var}=   Set Variable   ${your_variable}. In this case you use a "dummy" variable to push your data to the report files. You then don't call the Log function. This could however be confusing when looking at the testcase.

**Comment by mardukbp** [comment]
I had the same issue and solved it by creating a thread called "Main" and executing robot in that thread. I did it from a systray app that runs its GUI on its own thread.

**Comment by pekkaklarck** [comment]
This same problem was reported again as #5255. I fixed it and it will be part of RF 7.2.
